src/Type/BasePickerType.php

```diff
@@ -75,15 +79,23 @@ public function __construct(MomentFormatConverter $formatConverter, $translator,
 
         if ($translator instanceof LegacyTranslatorInterface) {
             $this->locale = $this->translator->getLocale();
-        } else {
+        } elseif ($translator instanceof TranslatorInterface) {
             $this->locale = $this->getLocale($requestStack);
+        } else {
+            /*
+             * NEXT_MAJOR: remove this check
+             */
+            @trigger_error(
```

Comment:
Why we trigger a deprecation here and not in other cases? (like relaxed argument and return types, classes marked as final, etc).

Reply:
Like i wrote some comments higer 0.x should be tha same as 1.x without return Types hints and with support for legacy, core code.

Reply:
I'm not saying the code should not be compatible. I guess the diff you've made is fine.
I mean, we should add deprecations in every case we know we've added strictness in 1.x. So, we should communicate these changes with  deprecation messages. Something like [this](https://github.com/sonata-project/SonataAdminBundle/pull/6023/files#diff-61e6d1f8fafe87b868adbe28ce0ba8f6R582).

Reply:
I want to keep it as close as possibile to 1.x. to contribute both version until sonata 3 is support. Anyway you are right. I will check it in Monday.